### PR TITLE
fix(es/module): handle this in class method

### LINKED ISSUE #5063: `this` in class method default parameter transpiles to `(void 0)`
### Describe the bug

Since `1.2.206`, when `module` is not set to `es6`, if a class with a method that has a default parameter references a class field with `this`, that reference will be transpiled to `(void 0)`, yielding an error when the code is run.

`this` is correctly retained when `module` is set to `es6` but is transpiled to `(void 0)` for all other values of `module`. Prior to `1.2.206`, `this` is retained regardless of the value of `module`.

### Input code

```typescript
class Foo {
    bar = 5
    getThing(a, b = this.bar) {
        return a + b;
    }
}
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript",
      "jsx": false
    },
    "target": "es2022",
    "loose": false,
    "minify": {
      "compress": false,
      "mangle": false
    }
  },
  "module": {
    "type": "commonjs"
  },
  "minify": false,
  "isModule": true
}
```

### Playground link

https://play.swc.rs/?version=1.2.207&code=H4sIAAAAAAAAA0vOSSwuVnDLz1eo5lIAgqTEIgVbBVMwOz21JCQjMy9dI1FHIQkoWpKRWawHVKAJVQsCRaklpUV5CokK2gpJ1mDRWq5aLgCgdCWCVgAAAA%3D%3D&config=H4sIAAAAAAAAA0WMTQrEIAyF75K1i8Fl79BDBCctFv9IUqiIdx8Fh64S3vve1%2BASB1uDgizE85OaFB%2FYgFxEceyLghnYiA4MQt2AIp%2BkExH7sXbUIWehBRiIPvmjTpnLsTCJvBWmM%2FzJPlwxf%2B8ZNNBaxp2TmNMl0F%2FRGnvZF618U%2F8BElOxt70AAAA%3D

### Expected behavior

The transpiled version of the above input code should retain the reference to `this`:

```js
"use strict";
class Foo {
    bar = 5;
    getThing(a, b = this.bar) {
        return a + b;
    }
}
```

### Actual behavior

The reference to `this` is transpiled to `(void 0)`, such that an error is yielded if the method is called:

```js
"use strict";
class Foo {
    bar = 5;
    getThing(a, b = (void 0).bar) {
        return a + b;
    }
}
```

### Version

1.2.207

### Additional context

On the [the MDN page for default parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters), one of the examples uses `this` in a default parameter position -- I don't think it's rare to do this.

### PR DESCRIPTION
**Pull request by magic-akari**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**
- fix #5063

### CODE REVIEW

File: crates/swc_ecma_transforms_module/src/module_ref_rewriter.rs [hunk 1]
```diff
@@ -101,36 +101,42 @@ impl VisitMut for ModuleRefRewriter {
     fn visit_mut_function(&mut self, n: &mut Function) {
         n.params.visit_mut_with(self);
```
Comment (Austaras) [inline crates/swc_ecma_transforms_module/src/module_ref_rewriter.rs:102]: param is not top level.
[babel](https://babeljs.io/repl#?browsers=chrome%2055&build=&builtIns=entry&corejs=3.6&spec=false&loose=false&code_lz=KYDwDg9gTgLgBAMwK4DsDGMCWEWIhACgEM4BeOGAC0wGcBKOAbzRxogBtgA6diAc2J0AvgCggA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.18.7&externalPlugins=&assumptions=%7B%7D)

Comment (magic-akari) [reply crates/swc_ecma_transforms_module/src/module_ref_rewriter.rs:102]: Thanks

Comment (Austaras) [reply crates/swc_ecma_transforms_module/src/module_ref_rewriter.rs]: visit_mut_class_method could be removed now.

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_transforms_module

**Pull request merged**